core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -880,6 +881,58 @@ public synchronized void removeQuorumServer(NetAddress serverNetAddress) throws
     }
   }
 
+  /**
+   * Transfers the leadership of the quorum to another server.
+   *
+   * @param newLeaderNetAddress the address of the server
+   * @throws IOException if error occurred while performing the operation
+   */
+  public synchronized void transferLeadership(NetAddress newLeaderNetAddress) throws IOException {
+    InetSocketAddress serverAddress = InetSocketAddress
+            .createUnresolved(newLeaderNetAddress.getHost(), newLeaderNetAddress.getRpcPort());
+    List<RaftPeer> oldPeers = new ArrayList<>(mRaftGroup.getPeers());
+    // The NetUtil function is used by Ratis to convert InetSocketAddress to string
+    String strAddr = NetUtils.address2String(serverAddress);
+    // if you cannot find the address in the quorum, throw exception.
+    if (oldPeers.stream().map(RaftPeer::getAddress).noneMatch(addr -> addr.equals(strAddr))) {
+      throw new IOException(String.format("<%s> is not part of the quorum <%s>.",
+              strAddr, oldPeers.stream().map(RaftPeer::getAddress).collect(Collectors.toList())));
+    }
+
+    RaftPeerId newLeaderPeerId = RaftJournalUtils.getPeerId(serverAddress);
+    // --- the change in priorities seems to be necessary otherwise the transfer fails ---
+    List<RaftPeer> peersWithNewPriorities = new ArrayList<>();
+    for (RaftPeer peer : oldPeers) {
+      peersWithNewPriorities.add(
+              RaftPeer.newBuilder(peer)
+              .setPriority(peer.getId().equals(newLeaderPeerId) ? 2 : 1)
+              .build()
+      );
+    }
+    // --- end of updating priorities ---
+    try (RaftClient client = createClient()) {
+      LOG.info("Applying new peer state before transferring leadership: <{}>",
+              peersWithNewPriorities);
```

Comment:
Need to convert `peersWithNewPriorities` to a string.

Reply:
This will call AbstractCollection#toString, the implementation is 
```java
    public String toString() {
        Iterator<E> it = iterator();
        if (! it.hasNext())
            return "[]";

        StringBuilder sb = new StringBuilder();
        sb.append('[');
        for (;;) {
            E e = it.next();
            sb.append(e == this ? "(this Collection)" : e);
            if (! it.hasNext())
                return sb.append(']').toString();
            sb.append(',').append(' ');
        }
    }
```

For RaftPeer#toString is 

```java
  public String toString() {
    String rpc = this.address != null ? "|rpc:" + this.address : "";
    String admin = this.adminAddress != null && !Objects.equals(this.address, this.adminAddress) ? "|admin:" + this.adminAddress : "";
    String client = this.clientAddress != null && !Objects.equals(this.address, this.clientAddress) ? "|client:" + this.clientAddress : "";
    String data = this.dataStreamAddress != null ? "|dataStream:" + this.dataStreamAddress : "";
    String p = "|priority:" + this.priority;
    return this.id + rpc + admin + client + data + p;
  }
```

FYI.